cns/service/main.go

```diff
@@ -187,6 +216,38 @@ func printVersion() {
 	fmt.Printf("Version %v\n", version)
 }
 
+// Try to register node with DNC when CNS is started in managed DNC mode
+func registerNode(httpRestService restserver.HTTPService, dncEP, infraVnet, nodeID string) {
+	var (
+		numCPU   = runtime.NumCPU()
+		response *http.Response
+		err      = fmt.Errorf("")
+		body     bytes.Buffer
+		httpc    = acn.GetHttpClient()
+	)
+
+	for err != nil {
+		response, err = httpc.Post(fmt.Sprintf("%s/%s/node/%s/cores/%d%s", dncEP, infraVnet, nodeID, numCPU, dncApiVersion), "application/json", &body)
+		if err == nil {
+			if response.StatusCode == http.StatusCreated {
+				var req cns.SetOrchestratorTypeRequest
+				json.NewDecoder(response.Body).Decode(&req)
+				httpRestService.SetNodeOrchestrator(&req)
+			} else {
+				logger.Errorf("[Azure CNS] Failed to register node %s/%s with managed DNC with http status code %s", infraVnet, nodeID, strconv.Itoa(response.StatusCode))
+			}
+
+			response.Body.Close()
+			time.Sleep(time.Second * 5)
+		} else {
+			logger.Errorf("[Azure CNS] Failed to register node %s/%s with err: %+v", infraVnet, nodeID, err)
+			return
```

Comment:
Do you intend to return here? If you return, you are not retrying when err is nil. I think you added return here by mistake.

Reply:
removed